lucid/main_window.py

```diff
@@ -30,57 +30,53 @@ class LucidMainWindow(QMainWindow):
         self.setCorner(Qt.BottomRightCorner, Qt.RightDockWidgetArea)
-        # Add toolbar
+        # Central Widget
+        self.central_widget = QStackedWidget()
+        self.setCentralWidget(self.central_widget)
+        # Toolbar
+        self.toolbar = LucidToolBar()
         self.addToolBar(Qt.TopToolBarArea, LucidToolBar())
 
-    def addDockWidget(self, area, dock):
-        """
-        Wrapped QMainWindow.addDockWidget
-
-        Add a QDockWidget to the LucidMainWindow. This is necessary in order to
-        force the tabbed behavior desired in the window. In addition it
-        performs the basic setup so the QDockWidget can be dragged to the
-        proper areas of the window, and also sets the focus on the dock so it
-        is immediately visible. This should rarely be called by external
-        users instead use the :meth:`.in_dock` decorator.
-
-        Parameters
-        ----------
-        area: Qt.DockWidgetArea
-
-        dock: QDockWidget
-        """
+    def setup_dock(self):
+        """Setup the qtpydocking system inside a standard Qt DockWidget"""
+        # If we've already loaded the docking system just return the active one
+        if self.main_dock:
+            return self.main_dock
+        # Docked DockWidget
+        self.main_dock = QDockWidget()
         # Force the dockwidget to only be allowed in areas determined by the
         # LucidMainWindow.allowed_docks
         allowed_flags = functools.reduce(operator.or_, self.allowed_docks)
-        dock.setAllowedAreas(allowed_flags)
-        super().addDockWidget(area, dock)
-        # If we already have an embedded dock, add it to existing dock. This
-        # needs to exist because even with ForceTabbedDocks the dock will be
-        # split if we add with regular API
-        embedded_docks = [curr_dock for curr_dock in self._docks
-                          if self.dockWidgetArea(curr_dock) == area
-                          and curr_dock.isVisible()]
-        if embedded_docks:
-            super().tabifyDockWidget(embedded_docks[0], dock)
-        # Raise to visible
-        dock.show()
-        dock.raise_()
-        dock.setFocus()
-
-    @property
-    def _docks(self):
-        """QDockWidget children"""
-        return [widget for widget in self.children()
-                if isinstance(widget, QDockWidget)]
+        self.main_dock.setAllowedAreas(allowed_flags)
+        # Place the dockmanager inside the dock
+        self.dock_manager = qtpydocking.DockManager(self.main_dock)
+        self.main_dock.setWidget(self.dock_manager)
+        self.main_dock.installEventFilter(self)
+        # Add to the first allowed location
+        self.addDockWidget(self.allowed_docks[0], self.main_dock)
+        return self.main_dock
+
+    def eventFilter(self, obj, event):
+        """Qt eventFilter to watch for closures of the docking system"""
+        # If the user closes the docking system clean up our internal state
+        if (self.main_dock
```

Comment:
Hmm, I think I should try to do better on the qtpydocking side, emitting a signal on close. While I don't have numbers, I'm fairly certan qt event filters are very pricey in Python.

The `if` block indentation looks off here, by the way...

Reply:
1. Hadn't heard about the cost of the `eventFilter` before, makes sense though if every event has to go through the filter... What's tricky about this (and this code in general), is this is a `PyQt.QDockWidget` not a `qtpydocking.DockWidget`, this block is watching for closing the whole dock system (and firing a reset)

Reply:
Alternative method in https://github.com/pcdshub/lucid/pull/2/commits/da5bfcf2a9b95079dd63e360eb9ec0c9fc4b870d

Reply:
Alternative method in https://github.com/pcdshub/lucid/pull/2/commits/da5bfcf2a9b95079dd63e360eb9ec0c9fc4b870d